dev/NavigationView/NavigationViewItemPresenter.cpp

```diff
@@ -138,3 +146,32 @@ void NavigationViewItemPresenter::UpdateCompactPaneLength(double compactPaneLeng
         }
     }
 }
+
+void NavigationViewItemPresenter::OnSplitViewPropertyChanged(const winrt::DependencyObject& /*sender*/, const winrt::DependencyProperty& /*args*/)
+{
+    UpdateClosedCompactVisualState();
+}
+
+void NavigationViewItemPresenter::UpdateClosedCompactVisualState()
+{
+    if (const auto navigationViewItem = GetNavigationViewItem())
```

Comment:
>isClosedCompact [](start = 23, length = 15)

Instead of being registered for splitview events in both the NavigationViewItem and NavigationViewItemPresenter, it might be better to just call this function from NavigationViewItem. This way we can keep the NavigationViewItemPresenter implementation as simple as possible.

Reply:
Great Suggestion!